src/app/public/modules/affix/dom-utils.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  SkyAffixAdapterCoords
+} from './affix-adapter-coords';
+
+/**
+ * Validates coordinates are fully visible within an element.
+ * @param parent
+ * @param coords
+ */
+export function verifyCoordsVisibleWithinElement(
+  parent: HTMLElement,
+  coords: SkyAffixAdapterCoords
+): boolean {
+  const parentCoords = getParentCoords(parent);
+  return !(
+    parentCoords.top > coords.top ||
+    parentCoords.right < coords.right ||
+    parentCoords.bottom < coords.bottom ||
+    parentCoords.left > coords.left
+  );
+}
+
+export function getScrollableParents(subject: HTMLElement): HTMLElement[] {
+  const bodyElement = window.document.body;
+  const result = [bodyElement];
+
+  let parentElement = subject.parentNode;
+
+  while (
+    parentElement !== undefined &&
+    parentElement !== bodyElement &&
+    parentElement instanceof HTMLElement
+  ) {
+    const overflowY = window.getComputedStyle(parentElement, undefined).overflowY.toLowerCase();
+    if (
+      overflowY === 'auto' ||
+      overflowY === 'hidden' ||
+      overflowY === 'scroll'
+    ) {
+      result.push(parentElement);
+    }
+
+    parentElement = parentElement.parentNode;
+  }
+
+  return result;
+}
+
+export function getImmediateScrollableParent(scrollableParents: HTMLElement[]): HTMLElement {
+  return scrollableParents[scrollableParents.length - 1];
+}
+
+export function getParentCoords(parent: HTMLElement): SkyAffixAdapterCoords {
```

Comment:
I'm guessing it has to do with the `document.body` check, but Is there something specific to this method that has different logic because its expecting a "parent" ? Why not just call it `getElementCoords` ?

Reply:
Good observation.